### PR TITLE
adding in fuzzySearch filter

### LINKED ISSUE #53: Add fuzzy search
Search results should be shown even when there is slight typo in search keyword.

### PR DESCRIPTION
**Pull request by LoganArnett**
# Related Issue	
Closes #53 

# Changes Proposed	
Added in a custom filter that takes in the search term and returns based on a `fuzzy` search so that there can be minor typos and still get the correct posts

### CODE REVIEW

File: public/fuzzySearch.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+          index[letter].push(i);
+        }
+      }
+    }
+  }
+  return index;
+};
+
+FuzzySearch.prototype.createIndex2 = function () {
+  var items = this.lowerCasedItems;
```
Comment (frankcash) [inline public/fuzzySearch.js]: can we move this to es6 const/let

File: public/fuzzySearch.js [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  };
+}
+
+angular.module("app").filter('fuzzy', fuzzySearchFilterFactory);
+
+function FuzzySearch(items, dataKey) {
+  this.items = items;
+  this.length = items.length;
+  this.lowerCasedItems = [];
+  for (var i = 0; i < this.length; i++) {
```
Comment (frankcash) [inline public/fuzzySearch.js]: can we use `for..in` or `for..each`

File: public/fuzzySearch.js [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+  this.index = this.createIndex();
+  this.index2 = this.createIndex2();
+  this.dumpIndex();
+  this.dumpIndex2();
+}
+
+FuzzySearch.prototype.createIndex = function () {
+  var items = this.lowerCasedItems;
+  var length = this.length;
+  var i = 0, j = 0;
```
Comment (frankcash) [inline public/fuzzySearch.js]: not sure I appreciate initializing these variables outside of the loop

**Comment by LoganArnett** [comment]
Updated based on review, please let me know if there are further requests

**Comment by frankcash** [comment]
Would this conflict with #68 ?

Also, this appears to break travis, could you pull master and then make any changes?

**Comment by frankcash** [comment]
This is very stale

**Pull request closed without merging**